File: cpp/1678_GoalParserInterpretation.cpp
class Solution {
public:
    unordered_map<string, string> map = {{"()", "o"}, {"(al)", "al"}};
    string interpret(string command) {
        deque<char> q;
        string curr = "";
        string res = "";
        for (char c : command) {
            if (c == ')') {
                while (!q.empty()) {
                    curr += q.front();
                    q.pop_front();
                }
                curr += ")";
                res += map[curr];
                curr = "";
            } else if (c == 'G') {
                res += c;
            } else {
                q.push_back(c);
            }
        }
        return res;
    }
};

File: swift/1678_GoalParserInterpretation.swift
class Solution {
    func interpret(_ command: String) -> String {
        var i = 0
        var idx = command.startIndex
        var result = ""
        while i < command.count {
            if command[idx] == "G" {
                result.append("G")
                i += 1
                idx = command.index(after: idx)
            } else if command[command.index(after: idx)] == ")" {
                result.append("o")
                i += 2
                idx = command.index(idx, offsetBy: 2)
            } else {
                result.append("al")
                i += 4
                idx = command.index(idx, offsetBy:4)
            }
        }
        return result
    }
}
